Enable parquet filter pushdown by default

alamb opened the pull request:
## Which issue does this PR close?

- related to #18873
- related to https://github.com/apache/datafusion/issues/3463 

## Rationale for this change

I am testing where we currently are with performance

## What changes are included in this PR?

<!--
There is no need to duplicate the description in the issue here but it is sometimes worth providing a summary of the individual changes in this PR.
-->

## Are these changes tested?

<!--
We typically require tests for all PRs in order to:
1. Prevent the code from being accidentally broken by subsequent changes
2. Serve as another way to document the expected behavior of the code

If tests are not included in your PR, please explain why (for example, are they covered by existing tests)?
-->

## Are there any user-facing changes?

<!--
If there are user-facing changes then we may require documentation to be updated before approving the PR.
-->

<!--
If there are any breaking changes to public APIs, please add the `api change` label.
-->

alamb committed:
Enable pushdown by defauly

alamb commented:
run benchmarks

adriangb commented:
What a 🎄 🎁 ! I see mostly speedups, vastly outweighing the slowdowns, but there are still slowdowns... any idea what those cases have in common? Any way I can help brainstorm solutions?

alamb commented:
> What a 🎄 🎁 ! I see mostly speedups, vastly outweighing the slowdowns, but there are still slowdowns... any idea what those cases have in common? Any way I can help brainstorm solutions?

Thanks @adriangb  -- I am not working full time the next few days but I am hoping to work on this particular item as it is something I desparately want to be done with (and I find it fun!)

I spent some time analyzing where the time is going. I haven't written up my notes yet, but I will do so (probably tomorrow as I have a long train ride). 

I found a few places to improve, but nothing that will obviously get back all the time (yet). I'll kep at it

Dandandan commented:
```
--------------------
Benchmark tpch_sf1.json
--------------------
┏━━━━━━━━━━━━━━┳━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Query        ┃      HEAD ┃ alamb_enable_pushdown ┃       Change ┃
┡━━━━━━━━━━━━━━╇━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━┩
│ QQuery 1     │ 190.82 ms │             199.11 ms │    no change │
│ QQuery 2     │  93.77 ms │             128.85 ms │ 1.37x slower │
│ QQuery 3     │ 125.10 ms │             192.31 ms │ 1.54x slower │
│ QQuery 4     │  75.03 ms │             124.79 ms │ 1.66x slower │
│ QQuery 5     │ 165.04 ms │             342.06 ms │ 2.07x slower │
│ QQuery 6     │  66.31 ms │             107.95 ms │ 1.63x slower │
│ QQuery 7     │ 208.69 ms │             250.14 ms │ 1.20x slower │
│ QQuery 8     │ 164.10 ms │             279.78 ms │ 1.70x slower │
│ QQuery 9     │ 222.55 ms │             389.32 ms │ 1.75x slower │
│ QQuery 10    │ 179.97 ms │             267.35 ms │ 1.49x slower │
│ QQuery 11    │  72.19 ms │             102.21 ms │ 1.42x slower │
│ QQuery 12    │ 116.14 ms │             258.92 ms │ 2.23x slower │
│ QQuery 13    │ 213.12 ms │             209.43 ms │    no change │
│ QQuery 14    │  92.02 ms │             102.09 ms │ 1.11x slower │
│ QQuery 15    │ 119.53 ms │             167.51 ms │ 1.40x slower │
│ QQuery 16    │  56.09 ms │              89.96 ms │ 1.60x slower │
│ QQuery 17    │ 263.36 ms │             278.76 ms │ 1.06x slower │
│ QQuery 18    │ 313.29 ms │             660.48 ms │ 2.11x slower │
│ QQuery 19    │ 132.03 ms │             169.52 ms │ 1.28x slower │
│ QQuery 20    │ 124.02 ms │             147.16 ms │ 1.19x slower │
│ QQuery 21    │ 253.64 ms │             308.28 ms │ 1.22x slower │
│ QQuery 22    │  43.38 ms │              65.67 ms │ 1.51x slower │
└──────────────┴───────────┴───────────────────────┴──────────────┘
```

This doesn't look great....

Dandandan commented:
run benchmark tpch

alamb committed:
Merge remote-tracking branch 'apache/main' into alamb/enable_pushdown

Dandandan committed:
Merge branch 'main' into alamb/enable_pushdown

Dandandan commented:
run benchmarks

Dandandan committed:
Merge branch 'main' into alamb/enable_pushdown

Dandandan commented:
run benchmarks

Dandandan committed:
Change config

Dandandan committed:
Change config

Dandandan committed:
Set enable_aggregate_dynamic_filter_pushdown default to true

Changed default value of enable_aggregate_dynamic_filter_pushdown from false to true.

Dandandan commented:
run benchmark tpch

Dandandan committed:
Decrease minimum file size for repartitioning
